lib/train/platforms/detect/helpers/os_common.rb

```diff
@@ -23,7 +23,11 @@ def unix_file_contents(path)
       # keep a log of files incase multiple checks call the same one
       return @files[path] if @files.key?(path)
 
-      res = @backend.run_command("test -f #{path} && cat #{path}")
+      if ARGV.include? "--sudo"
```

Comment:
Hrm, I'd like to find a better way of doing that - reaching back to `ARGV` here feels bad and brittle. Do we have access to the Connection options anywhere, perhaps within the `@backend` object?